src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -120,6 +122,7 @@ public void execute(SearchContext context) throws ElasticsearchException {
                 query = new FilteredQuery(query, searchFilter);
             }
             try {
+                collector.preCollection();
```

Comment:
ah, ok, I missed that is was only for the global aggs. Maybe it would be a good idea to rename this collector to something like `globalsCollector` for clarity?